scripts/oneoff/migrate-supplier-data-from-declarations.py

```diff
@@ -0,0 +1,200 @@
+
+Positional arguments:
+    <stage>                                 API stage to perform operation on
+    <data_api_token>                        appropriate API token for <stage>
+    <ch_api_token>                          API token for companies house API
+
+Optional arguments:
+    -h, --help                              show this help message and exit
+    --frameworks=<framework_slugs>          comma-separated whitelist of framework slugs to inspect for source data
+    --dry-run                               skip upload of modifications
+"""
+import sys
+sys.path.insert(0, '.')
+
+import re
+from pprint import pformat
+import logging
+import getpass
+
+import backoff
+from docopt import docopt
+import requests
+from dateutil.parser import parse as dateutil_parse
+from dmapiclient import DataAPIClient, HTTPError
+from dmapiclient.errors import HTTPTemporaryError
+
+from dmscripts.helpers.env_helpers import get_api_endpoint_from_stage
+from dmscripts.helpers.logging_helpers import configure_logger
+from dmscripts.helpers.logging_helpers import INFO as loglevel_INFO
+
+
+logger = logging.getLogger("script")
+
+
+_backoff_wrap = backoff.on_exception(
+    backoff.expo,
+    (HTTPTemporaryError, requests.exceptions.ConnectionError),
+    max_tries=5,
+)
+
+
+@_backoff_wrap
+def _chid_exists(chid, chapi_token):
+    response = requests.get(
+        "https://api.companieshouse.gov.uk/company/{}".format(chid),
+        auth=(chapi_token, ""),
+    )
+    if response.status_code == 404:
+        logger.info("chid %s does not seem to exist", repr(chid))
+        return False
+    elif response.status_code == 200:
+        logger.info("chid %s seems to exist", repr(chid))
+        return True
+    else:
+        raise RuntimeError("Requesting CHAPI page for chid %s resulted in a %s response" % (chid, response.status_code))
+
+
+_only_digits_alpha_re = re.compile("[^0-9A-Z]")
+_chid_validation_re = re.compile("^([0-9]{2}|[A-Z]{2})[0-9]{6}$")
+
+
+def _normalize_chid(in_data):
+    if not in_data:
+        return None
+    stripped = _only_digits_alpha_re.sub("", in_data.upper())
+    padded = ("O" + stripped) if stripped[0] == "C" and len(stripped) == 7 else stripped.rjust(8, "0")
+    return padded if _chid_validation_re.match(padded) else None
+
+
+def _catch_404_none(func):
+    try:
+        return func()
+    except HTTPError as e:
+        if e.status_code == 404:
+            return None
+        else:
+            raise
+
+
+if __name__ == '__main__':
+    arguments = docopt(__doc__)
+
+    configure_logger({"script": loglevel_INFO})
+
+    client = DataAPIClient(get_api_endpoint_from_stage(arguments['<stage>']), arguments['<data_api_token>'])
+    user = arguments['<user>'] or getpass.getuser()
+    framework_slugs = arguments.get("--frameworks") and arguments["--frameworks"].split(",")
+    dry_run = bool(arguments.get("--dry-run"))
+
+    frameworks = tuple(
+        fw for fw in sorted(
```

Comment:
I reckon if a list of frameworks was passed in on the command line then preserve the ordering there.  Or even just hard-code the framework list with the order we want in the script for this one-off use.